sound/soc/sof/intel/hda-dai.c

```diff
@@ -372,7 +373,8 @@ static int hda_link_hw_free(struct snd_pcm_substream *substream,
 		return -EINVAL;
 
 	stream_tag = hdac_stream(link_dev)->stream_tag;
-	snd_hdac_ext_link_clear_stream_id(link, stream_tag);
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		snd_hdac_ext_link_clear_stream_id(link, stream_tag);
```

Comment:
maybe it is better
```
if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
    stream_tag = hdac_stream(link_dev)->stream_tag;
    snd_hdac_ext_link_clear_stream_id(link, stream_tag);
}
```
like other codes in the driver.

Reply:
yes, I will refine it.